docs/contracts/visibility-and-getters.rst

```diff
@@ -142,8 +142,7 @@ it evaluates to a state variable.  If it is accessed externally
     }
 
 If you have a ``public`` state variable of array type, then you can only retrieve
-single elements of the array via the generated getter function. This mechanism
-exists to avoid high gas costs when returning an entire array. You can use
+single elements of the array via the generated getter function. You can use
```

Comment:
Note that you can call this function not only via `eth_call`, but also from other contracts. There it does cost gas.

Reply:
In any case, we should explain why it is like that somehow... :)

Reply:
Ok, thanks !!!

Reply:
Tracked in https://github.com/ethereum/solidity/issues/7565.